src/components/OTP.vue

```diff
@@ -0,0 +1,337 @@
+    enterOTPInputBoxDisplayText() {
+      return this.programData.text.default.enterOTP;
+    },
+
+    /** Returns the text for the button to request an OTP */
+    requestOTPButtonDisplayText() {
+      return this.programData.text.default.requestOTP;
+    },
+
+    /** Returns the text for the submit button */
+    submitButtonDisplayText() {
+      return this.programData.text.default.submitButton;
+    },
+
+    /** Returns the maximum length of the ID */
+    maxLengthOfId() {
+      return this.programData.input.maxLengthOfId;
+    },
+
+    /** Returns the basic validation type for the input */
+    basicValidationType() {
+      return this.programData.input.basicValidationType;
+    },
+
+    /** Returns the invalid input message stored against each program */
+    invalidInputText() {
+      return this.programData.text.default.invalid.input;
+    },
+  },
+  methods: {
+    /** Determines how the input box should look.
+     * - If an input error needs to be displayed, the box has a a red border.
+     * - Otherwise, it has an opacity of 30.
+     */
+    selectInputBoxClasses() {
+      let baseStyle =
+        "border-2 rounded-sm p-4 mx-auto border-gray-500 focus:border-gray-800 focus:outline-none";
+      return this.invalidInputMessage
+        ? baseStyle + "border-red-600 focus:border-red-600"
+        : baseStyle + "pointer-events-none opacity-30";
+    },
+
+    /** Calls the mapping function to validate the typed character
+     * @param {Object} event - event triggered when a character is typed
+     */
+    isValidPhoneNumber(event) {
+      if (validationTypeToFunctionMap[this.basicValidationType](event)) {
+        return true;
+      } else event.preventDefault();
+    },
+
+    /** Resets the invalid input message */
+    resetInvalidInputMessage() {
+      this.invalidInputMessage = "";
+    },
+
+    /** This function is called whenever something is entered in the input box.
+     * It checks if the required number of characters are being typed.
+     * @param {Object} event - the event which triggered this function
+     */
+    updatephoneNumber(event) {
+      if (event.target.value.length == 0) {
+        this.invalidInputMessage = "";
+      } else if (event.target.value.length > this.maxLengthOfId) {
+        event.target.value = event.target.value.slice(0, this.maxLengthOfId);
+        this.phoneNumber = event.target.value.toString();
+      } else if (event.target.value.length < this.maxLengthOfId) {
+        this.invalidInputMessage = this.invalidInputText;
+      } else {
+        this.resetInvalidInputMessage();
+      }
+    },
+
+    /** Function that calls the API to send an OTP
+     * If the status is 200, the OTP has been sent.
+     * Otherwise, retry
+     */
+    async sendOTP() {
+      const response = await OTPAuth.sendOTP(parseInt(this.phoneNumber));
+      if (response.status == 200) {
+        this.displayOTPMessage = mapSendStatusCodeToMessage(response.status.toString());
```

Comment:
This line is common to both, so it can be taken out of the `if`. And `this.OTPSent` can be set in one line